### DISCUSSION TITLE
S3 Upload Form Field - schema type for upload

### DISCUSSION THREAD

**Discussion by dqbd** [Budibase/budibase#5071] [Help] [answered]
Hello! As attachment types are not supported by non-internal DBs, we've been trying to use the new S3 upload form field, but we've been unable to figure out what column type should be used to connect this field properly.

<img width="270" alt="image" src="https://user-images.githubusercontent.com/1443449/159907211-43c64396-dbfa-4e6e-b7d6-b004ff8f2037.png">

Judging by the source code, it seems like the schema type must be "s3upload"?

**Comment by mjashanks** [comment, marked as answer]
Hi @dqbd - I was just trying this out myself, and came up with the answer - use an "Attachment" column type. Then I realised that the whole reason for you using this was because you are using an external DB!

What you can do is manually type the field name. This will allow you to upload files to S3, however, it will not link it to the record - i.e. it's one-way. If you re-open the record, there's no way to read the file from S3 (unless using the Budibase internal DB)

**Reply by dqbd** [reply]
RIght, will look forward to adding Attachment support when using external DBs! Will stick with internal DB for now. Thanks!
